src/norm_oracles.jl

```diff
@@ -360,3 +360,64 @@ function compute_extreme_point(lmo::EllipsoidLMO, direction; v=nothing, kwargs..
     mul!(v, I, lmo.buffer, -scaling, true)
     return v
 end
+
+
+"""
+    OrderWeightNormLMO(weights,radius)
+    
+LMO with feasible set being the atomic ordered weighted l1 norm: https://arxiv.org/pdf/1409.4271
+
+```
+C = {x ∈ R^n, Ω_w(x) ≤ R} 
+```
+The weights are assumed to be positive.
+"""
+struct OrderWeightNormLMO{R,B,D} <: LinearMinimizationOracle
+    radius::R
+    mat_B::B
+    direction_abs::D
+end
+
+function OrderWeightNormLMO(weights, radius)
+    N = length(weights)
+    B = zeros(N)
+    s = zero(eltype(weights))
```

Comment:
```suggestion
    s = zero(eltype(weights))
    B = zeros(float(typeof(s)), N)
```

that way the type of B is adapted to the type of the weights